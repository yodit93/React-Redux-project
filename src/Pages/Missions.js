import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { fetchMissions } from '../Redux/Missions/missionsSlice';
import MissionList from '../Components/MissionLists';

const Missions = () => {
  const { missions, isLoading, error } = useSelector((store) => store.missions);
  const dispatch = useDispatch();
  useEffect(() => {
    if (missions.length === 0) {
      dispatch(fetchMissions());
    }
  }, [dispatch, missions.length]);
  return (
    <>
      {isLoading && <h2>Loadding...</h2>}
      {error && <h2>{error}</h2>}
      <table className="missions">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>Join</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <MissionList key={mission.id} mission={mission} />
          ))}
        </tbody>
      </table>
    </>
  );
};

export default Missions;
