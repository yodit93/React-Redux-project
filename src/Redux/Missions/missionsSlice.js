/* eslint-disable */
/* stylelint-disable */
import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { fetchMissions } from '../Redux/Missions/missionsSlice';
import MissionList from '../Components/MissionLists';

const Missions = () => {
  const { missions, isLoading, error } = useSelector((store) => store.missions);
  console.log('missions data', missions);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);
  return (
    <>
      {isLoading && <h2>Loadding...</h2>}
      {error && <h2>{error}</h2>}
      <table className="missions">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>Join</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <MissionList key={mission.id} mission={mission} />
          ))}
        </tbody>
      </table>
    </>
  );
};

export default Missions;
