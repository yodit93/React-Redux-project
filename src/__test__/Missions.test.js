import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Missions from '../Pages/Missions';

const mockStore = configureMockStore();
const store = mockStore({
  missions: {
    missions: [
      {
        id: '1',
        name: 'Falcon 1',
        description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009',
      },
    ],
    isLoading: false,
    error: null,
  },
});

describe('Missions', () => {
  it('should render the Missions component', () => {
    const tree = renderer.create(<Provider store={store}><Missions /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should have a list', () => {
    render(
      <Provider store={store}>
        {' '}
        <Missions />
      </Provider>,
    );
    const tableElement = screen.getByRole('table');
    expect(tableElement).toBeInTheDocument();
  });
});
