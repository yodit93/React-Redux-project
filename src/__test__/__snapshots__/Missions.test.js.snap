// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions should render the Missions component 1`] = `
Array [
  " ",
  <table
    className="missions"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Join
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="row"
      >
        <td
          className="miss-name"
        >
          Falcon 1
        </td>
        <td
          className="miss-descri"
        >
          The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009
        </td>
        <td>
          <button
            className="memb"
            type="button"
          >
            Not a member
          </button>
        </td>
        <td>
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
